mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -49,6 +49,20 @@ private[feature] trait QuantileDiscretizerBase extends Params
 
   /** @group getParam */
   def getNumBuckets: Int = getOrDefault(numBuckets)
+
+  /**
+   * Relative error (see approxQuantile documentation for description).  Must be >= 0.
+   * default: 0.01
+   * @group param
+   */
+  val relativeError = new DoubleParam(this, "relativeError", "The relative target precision " +
+    "for approxQuantile",
+    ParamValidators.gtEq(0.0))
+  setDefault(relativeError -> 0.01)
```

Comment:
@oliverpierson by using `setDefault` here, `getRelativeError` will always return the default... if we want the formula to apply, we need to remove `setDefault` (note the tests may need to be adjusted for the new default case, and we should add tests for default param calculation and setting the param).